### DISCUSSION TITLE
Automating the process of plotting trajectory on a custom map

### DISCUSSION THREAD

**Discussion by iFieldSmart-Tech** [stella-cv/stella_vslam#609] [Q&A] [answered]
Hi,

Our current workflow is we get the starting point of the video from the user, they select the starting point on the map when they start recording. The map is a custom image opened in leaflet.js using crs simple, it is actually a floorplan of a construction site. Once the slam process is done, we use the output to plot the trajectory on the floorplan/map using the starting point. The issue is we still need to manually adjust scale and rotation for each trajectory. Is there any way to automate these 2 values for the trajectory?

**Comment by Skwangles** [comment, marked as answer]
If I understand your question correctly, to 'automate' adjusting the scale and rotation of a trajectory to plot correctly onto a floorplan image. 

My first thought is that you need atleast 2 groundtruth values to workout the scale and rotation - the easiest to me would be a start and an endpoint on the image.

If you have both of those, you can just solve for whatever scale and rotation matches the start and endpoint of the trajectory to the points on the image.
